src/data_provider/src/ports/portSolarisWrapper.h

```diff
@@ -0,0 +1,317 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015, Wazuh Inc.
+ * March 17, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _PORT_SOLARIS_WRAPPER_H
+#define _PORT_SOLARIS_WRAPPER_H
+
+#include <unistd.h>
+#include <inet/mib2.h>
+
+#include "iportWrapper.h"
+#include "sharedDefs.h"
+#include "bits/stdc++.h"
+
+static const auto cTcp = "tcp";
+static const auto cUdp = "udp";
+static const auto cTcp6 = "tcp6";
+static const auto cUdp6 = "udp6";
+
+static const std::map<int32_t, std::string> STATE_TYPE =
+{
+    /* For tcpConnState and tcp6ConnState */
+    { MIB2_TCP_closed,      "closed"        },
+    { MIB2_TCP_listen,      "listen"        },
+    { MIB2_TCP_synSent,     "synSent"       },
+    { MIB2_TCP_synReceived, "synReceived"   },
+    { MIB2_TCP_established, "established"   },
+    { MIB2_TCP_finWait1,    "finWait1"      },
+    { MIB2_TCP_finWait2,    "finWait2"      },
+    { MIB2_TCP_closeWait,   "closeWait"     },
+    { MIB2_TCP_lastAck,     "lastAck"       },
+    { MIB2_TCP_closing,     "closing"       },
+    { MIB2_TCP_timeWait,    "timeWait"      },
+    { MIB2_TCP_deleteTCB,   "deleteTCB"     }
+};
+
+static const std::map<int, std::string> UDP_STATE =
+{
+    { MIB2_UDP_unbound,     "unbound"   },
+    { MIB2_UDP_idle,        "idle"      },
+    { MIB2_UDP_connected,   "connected" },
+    { MIB2_UDP_unknown,     "unknown"   }
+};
+
+class SolarisPortWrapper final : public IPortWrapper
+{
+        const mib2_tcpConnEntry_t* tcpEntry;
+        const mib2_tcp6ConnEntry_t* tcp6Entry;
+        const mib2_udpEntry_t* udpEntry;
+        const mib2_udp6Entry_t* udp6Entry;
+
+    public:
```

Comment:
Have you considered a C++ trick to avoid the four similar structs and their constructors? For example, `std::variant`

Reply:
making use of this alternative makes the code excessively complex, which is originally intended to be C